composition-js/src/merging/merge.ts

```diff
@@ -1810,79 +1857,64 @@ class Merger {
     //   definition is the intersection of all definitions (meaning that if there divergence in
     //   locations, we only expose locations that are common everywhere).
     this.mergeDescription(sources, dest);
-    if (sources.some((s) => s && this.isMergedDirective(s))) {
+
+    const definedSources = sources.filter((src): src is DirectiveDefinition => src !== undefined);
+    if (definedSources.some(s => s.locations.some(loc => executableDirectiveLocations.includes(loc)))) {
+      if (this.mergedDirectives().includes(dest.name)) {
+        const { locations } = getLocationsFromDirectiveDefs(definedSources);
+        dest.addLocations(...locations);
```

Comment:
A little strange to set `dest`'s location here and to have it set _again_ in `mergeExecutableDirectiveDefinition`. I would simply move `getLocationsFromDirectiveDefs` within `mergeExecutableDirectiveDefinition`, same as `mergeTypeSystemDirectiveDefinition` does.